src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.OrderDao;
+import com.dev.cinema.exeption.DataProcessingExeption;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+
+@Dao
+public class OrderDaoImpl extends GenericDaoImpl<Order> implements OrderDao {
+    @Override
+    public Order add(Order entity) {
+        return super.add(entity);
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery(
+                    "SELECT DISTINCT o FROM Order o "
+                            + "JOIN FETCH o.tickets "
+                            + "JOIN FETCH o.user "
```

Comment:
There is no need to fetch user
If we use `@ManyToOne` or _@OneToOne_ type of connection, the fetch type is **_EAGER_** by default

Reply:
If you don't use fetch you have additional query to DB, to fetch user if you use fetch its all in one select, you can check it in console while debugging it.

Reply:
let's remove fetching user here